File: app/src/main/java/com/cogu/spylook/model/Contacto.java
package com.cogu.spylook.model;


import java.time.LocalDate;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Contacto {
    private int id;
    private String nombre;
    private String nickMasConocido;
    private int edad;
    private int foto;
    private LocalDate fechaNacimiento;
    private String ciudad;
    private String estado;
    private String pais;

    public Contacto(int id, String nombre, String nickMasConocido, LocalDate fechaNacimiento, String ciudad, String estado, String pais) {
        this.id = id;
        this.nombre = nombre;
        this.nickMasConocido = nickMasConocido;
        this.fechaNacimiento = fechaNacimiento;
        this.edad = LocalDate.now().isAfter(
                fechaNacimiento.withYear(LocalDate.now().getYear())
                        .withDayOfYear(LocalDate.now().getDayOfYear() - 1))
                ? LocalDate.now().getYear() - fechaNacimiento.getYear()
                : LocalDate.now().getYear() - (fechaNacimiento.getYear() - 1);
        this.ciudad = ciudad;
        this.estado = estado;
        this.pais = pais;
    }
    public Contacto(String nombre, String nickMasConocido, LocalDate fechaNacimiento, String ciudad, String estado, String pais) {
        this.nombre = nombre;
        this.nickMasConocido = nickMasConocido;
        this.fechaNacimiento = fechaNacimiento;
        this.edad = LocalDate.now().isAfter(
                fechaNacimiento.withYear(LocalDate.now().getYear())
                        .withDayOfYear(LocalDate.now().getDayOfYear() - 1))
                ? LocalDate.now().getYear() - fechaNacimiento.getYear()
                : LocalDate.now().getYear() - (fechaNacimiento.getYear() - 1);
        this.ciudad = ciudad;
        this.estado = estado;
        this.pais = pais;
    }
}

File: domain/src/main/java/com/cogu/domain/model/Contacto.kt
package com.cogu.domain.model

import utils.DateUtils
import java.time.LocalDate

data class Contacto(
    override var idAnotable: Int = 0,
    override var nombre: String,
    var alias: String,
    var fechaNacimiento: LocalDate,
    var ciudad: String,
    var estado: String,
    var pais: String,
    var colorFoto: Int,
    var edad: Int = DateUtils.calcularEdad(fechaNacimiento)
) : Anotable(idAnotable, nombre) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Contacto

        if (idAnotable != other.idAnotable) return false
        if (colorFoto != other.colorFoto) return false
        if (edad != other.edad) return false
        if (nombre != other.nombre) return false
        if (alias != other.alias) return false
        if (fechaNacimiento != other.fechaNacimiento) return false
        if (ciudad != other.ciudad) return false
        if (estado != other.estado) return false
        if (pais != other.pais) return false

        return true
    }

    override fun hashCode(): Int {
        var result = idAnotable
        result = 31 * result + colorFoto
        result = 31 * result + edad
        result = 31 * result + nombre.hashCode()
        result = 31 * result + alias.hashCode()
        result = 31 * result + fechaNacimiento.hashCode()
        result = 31 * result + ciudad.hashCode()
        result = 31 * result + estado.hashCode()
        result = 31 * result + pais.hashCode()
        return result
    }
}